### ISSUE TITLE
Unsubmitted form detection issues (e.g. on YouTube)

### ISSUE THREAD

**Issue by trlkly** [rNeomy/auto-tab-discard#312]
It seems that the form unsubmitted form detection logic cannot handle situations where submitting the form does not refresh the page. One common example is the YouTube comments section. If you submit a comment, that video's page will never be unloaded. 

I propose an addition to the current logic: After the first key event, add a setInterval that will check the value of that element after X seconds. If the value is "", then set window.isReceivingFormInput to false and  clearInterval. (Alternatively, implement the same logic but with setTimeout). 

My example case is YouTube.com. I will write it out formally:

### Steps to reproduce:

1. Make sure Auto Tab Discard extension is installed and the option "Do not discard a tab when form changes are not yet submitted" is selected. 
2. Navigate to any YouTube video (e.g. YouTube Rewind 2018 https://www.youtube.com/watch?v=YbJOTdZBX1g). 
3. Start typing a comment. 
4. Submit comment, or delete all text in comment. 
5. open another tab then use Auto Tab Discard to discard all other tabs..

### Expected results:
YouTube video page will be discarded. 

### Actual results
YouTube video page is not be discarded. 

### System info:
Browser: Chrome 106.0.5249.61 (64-bit)
OS: Windows 10 Pro version 10.0.19044 (21H2, 64-bit)
Auto Tab Discard version: 0.6.3.2

**Comment by rNeomy** [comment]
Take a look at the new logic.
https://github.com/rNeomy/auto-tab-discard/blob/master/v3/data/inject/watch.js

1. It now stores all the edited elements, and ignores them if there is no value.
2. If a form is submitted on the page, it resets the state and deletes all the stored elements.

Please give it a try and let me know if it works.

**Comment by trlkly** [comment]
I take it I should just install the current version's v3 folder as an
unpacked extension? Or is there a testing version of the extension I could
install through the Chrome store?

**Comment by rNeomy** [comment]
No, there is no testing version. Just use the current master branch (v3 directory)

https://github.com/rNeomy/auto-tab-discard/archive/refs/heads/master.zip

**Comment by trlkly** [comment]
I have had no problems with it. It even works if I cancel the comment. I however, haven't tested on any other sites like this, as YouTube is the main reason I use your extension. 

(Chrome basically never discards YouTube tabs on its own, and so I kept running into issues where the tab would get cached out. This greatly slows down shutdown, at least on my system, since it seems Chrome reads the tabs back into memory before shutdown.)

**Comment by trlkly** [comment]
Oh, and BTW, my main reason for suggesting setInterval was because I was concerned about performance of checking while typing without some sort of time gate. But perhaps that was a premature performance optimization.

**Comment by trlkly** [comment]
Is this now available in the November 1 update in the Chrome Web Store? I'm looking to stop using the developer extension and go back to the real extension.

**Comment by rNeomy** [comment]
I'll submit the new version tomorrow.

**Comment by ErikAnderson3** [comment]
This issue appears to be related to https://github.com/rNeomy/auto-tab-discard/issues/339.

On that issue, a couple of us have run into the opposite problem — rather than Auto Tab Discard erroneously keeping a tab in memory, we're encountering cases where Auto Tab Discard erroneously discards a tab even though we are in the midst of entering text, and have opened other tabs to confirm things that we are writing about.
